polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5309,12 +5332,45 @@ bool MLIRASTConsumer::HandleTopLevelDecl(DeclGroupRef dg) {
     if (name == "cudaGetDevice" || name == "cudaMalloc")
       continue;
 
+    // Temp HACK: filter out SPIRV builtins that reference global variables.
```

Comment:
When a device function is put in the GPUModuleOp we will need to also place in the module any global variable declaration referenced by the function. This is just a workaround. That work can be done in a subsequent PR.